modules/extensions/src/collide/collide-effect.ts

```diff
@@ -0,0 +1,254 @@
+import {Texture2D, cssToDeviceRatio} from '@luma.gl/core';
+import {readPixelsToArray} from '@luma.gl/core';
+import type {Effect, Layer, PreRenderOptions, Viewport} from '@deck.gl/core';
+import CollidePass from './collide-pass';
+
+type CollideExtensionProps = {
+  collideTestProps?: {};
+  collideGroup: string;
+};
+
+// Factor by which to downscale Collide FBO relative to canvas
+const DOWNSCALE = 2;
+
+type RenderInfo = {
+  collideGroup: string;
+  layers: Layer<CollideExtensionProps>[];
+  layerBounds: ([number[], number[]] | null)[];
+};
+
+// Class to manage collide effect
+export default class CollideEffect implements Effect {
+  id = 'collide-effect';
+  props = null;
+  useInPicking = true;
+
+  private channels: Record<string, RenderInfo> = {};
+  private collidePasses: Record<string, CollidePass> = {};
+  private dummyCollideMap?: Texture2D;
+  private lastViewport?: Viewport;
+
+  preRender(
+    gl: WebGLRenderingContext,
+    {effects: allEffects, layers, layerFilter, viewports, onViewportActive, views}: PreRenderOptions
+  ): void {
+    if (!this.dummyCollideMap) {
+      this.dummyCollideMap = new Texture2D(gl, {width: 1, height: 1});
+    }
+
+    const collideLayers = layers.filter(
+      // @ts-ignore
+      ({props: {visible, extensions, collideGroup}}) =>
```

Comment:
It will be easier to just add a `collideEnabled` prop and default to `true` following the other extensions' convention.